.github/workflows/test-compilation.yml

```diff
@@ -0,0 +1,74 @@
+name: Test dd-trace-rb compilation
+
+on:
+  push:
+  workflow_call:
+
+jobs:
+  test-compilation:
+    runs-on: ubuntu-latest
+    name: "Ruby:${{ matrix.ruby_version }} expects ${{ matrix.expected_exit_code == 1 && '❌' || '✅' }} for ${{ matrix.ref }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Success cases (github.sha)
+          - { ruby_version: '3.4', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.3', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.2', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.1', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '3.0', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '2.7', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '2.6', ref: '${{ github.sha }}', expected_exit_code: 0 }
+          - { ruby_version: '2.5', ref: '${{ github.sha }}', expected_exit_code: 0 }
+
+          # Success cases (test-negative/missing-versions)
+          - { ruby_version: '3.2', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '3.1', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '3.0', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '2.7', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+          - { ruby_version: '2.6', ref: 'test-negative/missing-versions', expected_exit_code: 0 }
+
+          # Failure cases (test-negative/missing-versions)
+          - { ruby_version: '3.4', ref: test-negative/missing-versions, expected_exit_code: 1 }
+          - { ruby_version: '3.3', ref: test-negative/missing-versions, expected_exit_code: 1 }
+          - { ruby_version: '2.5', ref: test-negative/missing-versions, expected_exit_code: 1 }
```

Comment:
I'm not sure I fully understand why this is useful... Is this a "test the tests" kinda thing -- make sure we're actually testing correctly by verifying that if we remove the headers, our tests will flag it?

Since this branch needs to be manually updated, I'm not entirely sure it would catch issues related to CI setup though... One alternative I could suggest would be to delete only one of the headers, such as `iseq.h` or `vm_core.h` -- that could be done on the same branch and should always trigger a failure.

I would suggest as well at least reducing it to one Ruby version -- it's highly likely that if we break something, we'll affect all Rubies, not just one (and as I mentioned above, the dd-trace-rb gives test coverage for the full range of versions anyway, so it would flag and really really rare issues still)